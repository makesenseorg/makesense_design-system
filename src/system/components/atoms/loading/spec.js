import { shallowMount } from "@vue/test-utils";
import Comp from "./Loading.vue";

describe("Loading.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {},
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sets type modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        type: "full",
      },
    });
    expect(wrapper.classes()).toContain("loading--full");
  });

  it("sets inline modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        inline: true,
      },
    });
    expect(wrapper.classes()).toContain("loading--inline");
  });

  it("sets color modifier on svg icon", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        color: "#ffffff",
      },
    });
    expect(wrapper.find(".loading__svg").attributes("stroke")).toBe("#ffffff");
  });

  it("sets color modifier on svg icon circle", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        color: "#ffffff",
      },
    });
    expect(wrapper.find(".loading__svg circle").attributes("stroke")).toBe(
      "#ffffff"
    );
  });

  it("sets size width modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        size: 60,
      },
    });
    expect(wrapper.find(".loading__svg").attributes("width")).toBe("60");
  });

  it("sets size height modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        size: 60,
      },
    });
    expect(wrapper.find(".loading__svg").attributes("height")).toBe("60");
  });

  it("display progress container", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        progress: 50,
      },
    });

    expect(wrapper.find(".loading__progress").exists()).toBe(true);
  });

  it("display progress number", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        progress: 50,
      },
    });

    expect(wrapper.find(".loading__progress").text()).toBe("50%");
  });
});
