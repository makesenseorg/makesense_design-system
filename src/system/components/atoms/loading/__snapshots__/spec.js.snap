// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading.vue matches snapshot 1`] = `
<div
  class="loading loading--full"
>
  <svg
    class="loading__svg"
    height="50"
    stroke="#3c4858"
    viewBox="-3 -3 44 44"
    width="50"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <g
        stroke-width="3"
        transform="translate(1 1)"
      >
        <circle
          cx="18"
          cy="18"
          r="18"
          stroke="#3c4858"
          stroke-opacity="0.5"
        />
         
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
        >
          <animatetransform
            attributeName="transform"
            dur="1s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
   
  <!---->
</div>
`;
